### PR TITLE
Destination-MotherDuck: Fix bug in _flush_buffer, explicitly register buffer dataframe, fix bug in _execute_sql

### PR DESCRIPTION
**Pull request by guenp**
## What
- Fixes a bug in _flush_buffer that broke the UTs
- Add a method to explicitly register a buffer dataframe before querying it
- Fix bug in _execute_sql (results promise was returned after closing db connection without fetching rows)

## How
- _flush_buffer should use the correct stream name to write data and also check if the data exists in the buffer
- duckdb_engine lets you register dataframes using the `execute` method!
- _execute_sql now fetches the results before returning them and closing the db connection

## Review guide
1. airbyte-integrations/connectors/destination-motherduck/destination_motherduck/destination.py
2. airbyte-integrations/connectors/destination-motherduck/destination_motherduck/processors/duckdb.py
3. airbyte-integrations/connectors/destination-motherduck/integration_tests/integration_test.py

## User Impact
This fixes a blocking bug

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/destination-motherduck/destination_motherduck/processors/duckdb.py [hunk 1]
```diff
@@ -101,6 +103,46 @@ class DuckDBSqlProcessor(SqlProcessorBase):
...
+        Execute the given SQL statement.
+
+        Explicitly register a buffer table to read from.
+        """
+        if isinstance(sql, str):
+            sql = text(sql)
+
+        with self.get_sql_connection() as conn:
+            try:
+                # This table will now be queryable from DuckDB under the name BUFFER_TABLE_NAME
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-motherduck/destination_motherduck/processors/duckdb.py:134]: @guenp - Is this a permanent table addition or just for the session/query context? Assuming the latter, is there any risk of name collisions? I assume not, but just want to double check.

File: airbyte-integrations/connectors/destination-motherduck/integration_tests/integration_test.py [hunk 2]
```diff
@@ -303,6 +303,30 @@ def _state(data: Dict[str, Any]) -> AirbyteMessage:
     return AirbyteMessage(type=Type.STATE, state=AirbyteStateMessage(data=data))
 
 
+@pytest.fixture()
+def sql_processor(
+        configured_catalogue,
+        test_schema_name,
+        config: Dict[str, str]
+    ):
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-motherduck/integration_tests/integration_test.py:311]: Nice refactor. 👍

**Review by aaronsteers — APPROVED** [review]
LGTM! Just one question inline. ✅

File: airbyte-integrations/connectors/destination-motherduck/destination_motherduck/processors/duckdb.py (hunk 1 above)
Comment (guenp) [reply airbyte-integrations/connectors/destination-motherduck/destination_motherduck/processors/duckdb.py:134]: Great question. I believe it's only for the context!

**Pull request merged**